src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedNonNullTypesAttributeSymbol.cs

```diff
@@ -0,0 +1,172 @@
+            {
+                var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+
+                Binder.ReportUseSiteDiagnostics(boolType, diagnostics, Location.None);
+
+                var boolWithAnnotations = TypeSymbolWithAnnotations.Create(boolType);
+                // PROTOTYPE constructor should save the parameter into a field (for users of reflection)
+                return ImmutableArray.Create<MethodSymbol>(
+                    new NonNullTypesAttributeConstructorSymbol(
+                        containingType,
+                        m => ImmutableArray.Create(SynthesizedParameterSymbol.Create(m, boolWithAnnotations, 0, ConstantValue.True, name: "flag"))));
+            }
+        }
+
+        internal override bool RequiresCompletion
+            => true;
+
+        internal override bool HasComplete(CompletionPart part)
+            => _state.HasComplete(part);
+
+        internal override void ForceComplete(SourceLocation locationOpt, CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var incompletePart = _state.NextIncompletePart;
+                switch (incompletePart)
+                {
+                    case CompletionPart.Attributes:
+                        GetAttributes();
+                        break;
+
+                    case CompletionPart.None:
+                        return;
+
+                    default:
+                        // any other values are completion parts intended for other kinds of symbols
+                        _state.NotePartComplete(CompletionPart.All & ~CompletionPart.InjectedSymbolAll);
+                        break;
+                }
+
+                _state.SpinWaitComplete(incompletePart, cancellationToken);
+            }
+
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            var attributes = _lazyCustomAttributes;
+            if (!attributes.IsDefault)
+            {
+                return attributes;
+            }
+
+            if (ImmutableInterlocked.InterlockedInitialize(ref _lazyCustomAttributes, MakeAttributes()))
+            {
+                var completed = _state.NotePartComplete(CompletionPart.Attributes);
+                Debug.Assert(completed);
+            }
+
+            return _lazyCustomAttributes;
+        }
+
+        /// <summary>
+        /// Adds an `[AttributeUsage(AttributeTargets.Class | ...)]` (if possible) and captures any diagnostics in the process.
+        /// </summary>
+        private ImmutableArray<CSharpAttributeData> MakeAttributes()
+        {
+            var ctor = (MethodSymbol)DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_AttributeUsageAttribute__ctor);
```

Comment:
>GetWellKnownTypeMember [](start = 58, length = 22)

We probably have a helper in Binder that reports all required diagnostics as well. Consider using it instead of duplicating this logic here.